### DISCUSSION TITLE
can i verify data use randread

### DISCUSSION THREAD

**Discussion by huangyu96** [axboe/fio#1839] [Q&A] [answered]
![image](https://github.com/user-attachments/assets/a91eb4e3-8d1f-448a-abd5-67f95aa628e4)

1.
 /root/fio3.38/fio  --ioengine=io_uring --rw=randwrite --direct=1 --iodepth=6 --bs=1024 --do_verify=1 --offset=1024 --size=13k --verify=crc32 --filename=/dev/nvme0n1 --name=nvme0

2.
/root/fio3.38/fio --filename=/dev/nvme0n1 --numjobs=1 --bs=1024  --iodepth=6 --rw=randread --verify_state_load=1 --verify=crc32 --do_verify=1 --verify_fatal=1 --verify_dump=1  --direct=1 --thread=1 --group_reporting  --ioengine=io_uring --name=nvme0 --exitall_on_error=1

Why check the place of 8392577024,Are my parameters wrong? How can I verify randomly written data？
What I understand is that he loads local-nvme0-0-verify.state file to get the check range and other information, and does not need to follow the parameter offset

**Comment by vincentkfu** [comment, marked as answer]
The verify state file does not contain full details about the workload. To use it effectively you should run the same workload that was used to generate the state file. The main purpose of the state file is to tell fio when to stop verifying.

https://fio.readthedocs.io/en/latest/fio_doc.html#cmdoption-arg-verify_state_load
Some more details about the verify state file: https://github.com/vincentkfu/fio-blog/wiki/Testing-on-a-big-endian-platform

**Reply by huangyu96** [reply]
![image](https://github.com/user-attachments/assets/8f9c5d9b-b354-496a-bf1c-63cce0b5f7d6)
![image](https://github.com/user-attachments/assets/be6b7a67-a37d-4c93-9f95-9932c3cc25ff)
Can I perform high performance verification? Are there any parameters to solve this problem? Currently, I can only verify the SSD power failure scenario at iodepth=1 or iodepth_batch_complete_min=0.

**Reply by sitsofe** [reply]
@huangyu96 The original question ("can i verify data use randread") has been answered by Vincent ("only when you keep the parameters the same"). Your follow up question ("Can I perform high performance verification?") has been filed as #1853 so I'm going to close this discussion.
